src/main/java/com/dataguise/saas/service/ProtectionTaskService.java

```diff
@@ -0,0 +1,611 @@
+package com.dataguise.saas.service;
+
+import com.dataguise.saas.controllers.SettingsController;
+import com.dataguise.saas.dto.*;
+import com.dataguise.saas.exception.BadGatewayException;
+import com.dataguise.saas.exception.BadRequestException;
+import com.dataguise.saas.exception.InternalServerException;
+import com.dg.saas.orch.client.DgSecureRestDriver;
+import com.dg.saas.orch.exception.DgDrvException;
+import com.dg.saas.orch.models.builders.MaskerTaskBuilder;
+import com.dg.saas.orch.models.structures.DgMaskerStatus;
+import com.dg.saas.orch.models.structures.ErrorConstants;
+import com.dg.saas.orch.models.structures.Modules;
+import io.swagger.client.model.*;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Service
+public class ProtectionTaskService {
+
+    private static Logger logger = Logger.getLogger(ProtectionTaskService.class);
+
+    @Autowired
+    private DgSecureRestDriver restDriver;
+
+    @Autowired
+    private SettingsController settingsController;
+
+    @Autowired
+    private TaskBuilderService taskBuilderService;
+
+    /**
+     * Create a Protection task
+     * @param protectionTask
+     * @return
+     * @throws BadGatewayException
+     */
+    public Integer createTask(@RequestBody DgMaskerTaskDTO protectionTask, String sourceModule) throws BadGatewayException {
+        Integer taskId = null;
+        try {
+            taskId = restDriver.createMaskerTask(protectionTask, sourceModule);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Protection task created with id: " + taskId);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ProtectionTaskService.java L57
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a296048558>*